### PR TITLE
#120 TabsText variant support

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-components-react-native/issues/120 |
| Decisions | • Added variant prop to `TabsText` and `ButtonTabText` components <br>• `TabsText` and `ButtonTabText` components now support `compact` variant <br>• Fixed missing `style` prop in `ButtonTabText` component |
| Animated GIF | RN-TabsText_variant_compact |

### CODE REVIEW

File: react/components/atoms/button-tab-text/button-tab-text.js [hunk 1]
```diff
@@ -65,12 +71,20 @@ const styles = StyleSheet.create({
...
+    buttonTabTextCompact: {
+        paddingVertical: 9,
+        paddingHorizontal: 16
+    },
     text: {
         marginTop: Platform.OS === "ios" ? 4 : 0,
         fontFamily: baseStyles.FONT,
         fontSize: 16,
-        letterSpacing: 0.25,
-        textAlign: "center"
```
Comment (gcandal) [inline react/components/atoms/button-tab-text/button-tab-text.js]: Why?

File: react/components/molecules/tabs-text/tabs-text.js [hunk 2]
```diff
@@ -75,21 +80,31 @@ export class TabsText extends PureComponent {
...
-        return [styles.tabsText, this.props.style];
+        return [
+            styles.tabsText,
+            styles[`tabsText${capitalize(this.props.variant)}`],
+            this.props.style
+        ];
     }
 
+    _buttonStyle = () => {
+        return this.props.variant === undefined && styles.button;
```
Comment (gcandal) [inline react/components/molecules/tabs-text/tabs-text.js]: Do the same `capitalize` trick here. You're probably thinking something like "only apply style when it has no variant", but it must be "apply the general style and override with the variant".

File: react/components/atoms/button-tab-text/button-tab-text.js (hunk 1 above)
Comment (NFSS10) [reply react/components/atoms/button-tab-text/button-tab-text.js]: Didn't see any difference with and without. Checked in both iOS and Android

File: react/components/molecules/tabs-text/tabs-text.stories.js [hunk 3]
```diff
@@ -14,8 +14,16 @@ storiesOf("Molecules", module)
             { text: "New", disabled: false },
             { text: "Past", disabled: true }
         ];
+        const variant = select(
```
Comment (gcandal) [inline react/components/molecules/tabs-text/tabs-text.stories.js]: Add this to the `buttonTabText` too

**Pull request merged**